src/shmem_team.c

```diff
@@ -371,33 +371,41 @@ int shmem_internal_team_split_2d(shmem_internal_team_t *parent_team, int xrange,
     const int num_xteams = ceil( parent_size / (float)xrange );
     const int num_yteams = xrange;
 
-    int start = parent_start;
+    int start = 0;
     int ret = 0;
 
+    shmem_internal_team_t *my_xteam, *my_yteam;
+
     for (int i = 0; i < num_xteams; i++) {
         int xsize = (i == num_xteams - 1 && parent_size % xrange) ? parent_size % xrange : xrange;
 
         ret = shmem_internal_team_split_strided(parent_team, start, parent_stride,
-                                                xsize, xaxis_config, xaxis_mask, xaxis_team);
+                                                xsize, xaxis_config, xaxis_mask, &my_xteam);
         if (ret) {
-            RAISE_ERROR_MSG("Creation of x-axis team %d of %d failed\n", i, num_xteams);
+            RAISE_ERROR_MSG("Creation of x-axis team %d of %d failed\n", i+1, num_xteams);
         }
-        start += xrange * parent_stride;
+        start += xrange;
+
+        if (my_xteam != SHMEMX_TEAM_INVALID)
+            *xaxis_team = my_xteam;
```

Comment:
I wonder if we want to add an assert here to ensure that `xaxis_team` is not assigned to more than once? Also, you can move the declaration of `my_xteam` into the loop body above and `my_yteam` into the loop body below. It looks like neither should be referenced outside of the loop body.